src/material/form-field/form-field.ts

```diff
@@ -703,12 +703,17 @@ export class MatFormField
     const element: HTMLElement = this._elementRef.nativeElement;
     if (element.getRootNode) {
       const rootNode = element.getRootNode();
-      // If the element is inside the DOM the root node will be either the document
-      // or the closest shadow root, otherwise it'll be the element itself.
-      return rootNode && rootNode !== element;
+      // If the element is inside the DOM the root node will be either the document,
+      // the closest shadow root or an element that is not yet rendered, otherwise it'll be the element itself.
+      if (rootNode && rootNode !== element) {
+        if (rootNode === document || _getShadowRoot(element)) {
+          return true;
+        }
+      }
     }
     // Otherwise fall back to checking if it's in the document. This doesn't account for
     // shadow DOM, however browser that support shadow DOM should support `getRootNode` as well.
+    // If the element is not in the document it means that the element is not yet rendered.
     return document.documentElement!.contains(element);
```

Comment:
Actually I think we would have to update this as well. `contains` at the document level won't work if the element is in the shadow DOM, it has to be called on the shadow root.